io/net/bonding.py

```diff
@@ -283,19 +319,28 @@ def tearDown(self):
                                     self.host_ips, self.net_mask):
             cmd = "ifconfig %s %s netmask %s up"\
                   % (val1, val2, val3)
+            for i in range(0, 600, 60):
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("Unable to bring up %s interface\
+                                   in Host machine" % val1)
+                    if 'state UP' in process.system_output("ip link \
+                            show %s" % val1, shell=True):
+                        break
+                    time.sleep(60)
```

Comment:
I might have overlooked in previous PR. But I do not get what you are trying to do here.